packages/data/src/namespace-store/index.js

```diff
@@ -245,6 +246,7 @@ function mapResolvers( resolvers, selectors, store ) {
 
 				const { metadata } = store.__unstableOriginalGetState();
 				if (
+					running ||
```

Comment:
I'm not sure about this boolean, it seems like it can cause rase conditions if we trigger multiple resolvers/selectors one after the other and some of them will be ignored?

Reply:
I wonder if we should leverage "priority-queue" package instead.